File: Codigos/PUJ_ML_01_en_clase/examples/python3/LogisticModel_01.py
## =========================================================================
## @author Leonardo Florez-Valencia (florez-l@javeriana.edu.co)
## =========================================================================

import numpy, sys
sys.path.insert( 0, '../../lib/python3' )

from PUJ_ML_01_en_clase.lib.python3.PUJ.Model.Logistic import  Logistic
from PUJ_ML_01_en_clase.lib.python3.PUJ.Optimizer import *

# Load data
data = numpy.loadtxt( open( sys.argv[ 1 ], 'rb' ), delimiter = ',' )

m = Logistic( )
m.setParameters( [ 0 for i in range( data.shape[ 1 ] ) ] )
print( 'Initial model = ' + str( m ) )

# Configure cost
J = Logistic.Cost( m, data[ : , 0 : -1 ], data[ : , -1 : ] )

# Debugger
debugger = Debug.Simple

# Fit using an optimization algorithm
opt = GradientDescent( J )
opt.setDebugFunction( debugger )
opt.setLearningRate( 1e-2 )
opt.setNumberOfIterations( 1000 )
opt.setNumberOfDebugIterations( 10 )
opt.Fit( )

# Show results
print( '===========================================' )
print( '= Iterations       :', opt.realIterations( ) )
print( '= Fitted model     :', m )
print( '===========================================' )

"""
m.setParameters( [ 0, -1, 0 ] )
print( m )

print( m.evaluate( [ 2, 2 ] ) )
print( m.threshold( [ 2, 2 ] ) )

print( m.evaluate( [ -2, -1 ] ) )
print( m.threshold( [ -2, -1 ] ) )

print( m.evaluate( [ -10, 30 ] ) )
print( m.threshold( [ -10, 30 ] ) )
"""



## eof - $RCSfile$

File: PUJ_ML-3/examples/cxx/Logistic_Model_01.cxx
// =========================================================================
// @author Leonardo Florez-Valencia (florez-l@javeriana.edu.co)
// =========================================================================

#include <iostream>
#include <PUJ_ML/Model/Logistic.h>

using TScalar = double;
using TModel  = PUJ_ML::Model::Logistic< TScalar >;

int main( int argc, char** argv )
{
  // Curly initialization
  TModel m1( { 10, 1, 2, 3 } );
  std::cout << m1 << std::endl;
  TModel::TRow x1( m1.inputSize( ) );
  x1 << -1, -2, -3;
  std::cout << m1.evaluate( x1 ) << std::endl;
  std::cout << m1.threshold( x1 ) << std::endl;

  // Stream-based initialization
  TModel m2;
  m2 << 4 << 10 << 1 << 2 << 3;
  std::cout << m2 << std::endl;
  TModel::TRow x2( m2.inputSize( ) );
  x2 << -1, -2, -3;
  std::cout << m2.evaluate( x2 ) << std::endl;
  std::cout << m2.threshold( x2 ) << std::endl;

  return( EXIT_SUCCESS );
}

// eof - $RCSfile$
